eclipse/set_memory_flags.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import pathlib
+
+
+def remove_all_memory_flags(file_content: list[str]) -> list[str]:
+    """Remove all memory flags from *.ini files.
+
+    To make sure that we don't have any conflicting memory flags in the
+    .ini file, we remove all flags that start with -Xm.
+
+    This will explicitely remove the -Xms and -Xmx flags.
+    """
+
+    return [
+        line for line in file_content if not line.strip().startswith("-Xm")
+    ]
+
+
+def set_memory_flags(
+    file_content: list[str], memory_min: str, memory_max: str
+) -> None:
+    """Set the memory flags in *.ini.
+
+    If the value of memory_max ends with a %, we assume that it's a percentage value.
+    - If used in a Kubernetes cluster, it determines the values from the Pod requests/limits.
+    - If used on a host system, it determines the value from the host memory.
+
+    See Also
+    --------
+    - https://stackoverflow.com/a/65327769
+    - https://www.merikan.com/2019/04/jvm-in-a-container/#java-10
+    """
+
+    if memory_min.strip().endswith("%"):
+        append_flag_to_file(
+            file_content, "XX:InitialRAMPercentage=", memory_min.strip("%")
+        )
+        append_flag_to_file(
+            file_content, "XX:MinRAMPercentage=", memory_min.strip("%")
+        )
+    else:
+        append_flag_to_file(file_content, "Xms", memory_min)
+
+    if memory_max.strip().endswith("%"):
+        append_flag_to_file(
+            file_content, "XX:MaxRAMPercentage=", memory_max.strip("%")
+        )
+    else:
+        append_flag_to_file(file_content, "Xmx", memory_max)
+
+
+def print_vm_memory_usage_during_start(file_content: list[str]) -> None:
+    """Print the JVM memory options during start."""
+
+    file_content.append("-XshowSettings:vm")
+
+
+def append_flag_to_file(
+    file_content: list[str], flag: str, value: str
+) -> None:
+    """Append a flag to the *.ini file."""
+
+    file_content.append(f"-{flag}{value}")
+
+
+if __name__ == "__main__":
+    eclipse_executable = pathlib.Path(os.environ["ECLIPSE_EXECUTABLE"])
+    _memory_min = os.environ["MEMORY_MIN"]
+    _memory_max = os.environ["MEMORY_MAX"]
+
+    ini_path = eclipse_executable.with_suffix(".ini")
+
+    _file_content = ini_path.read_text("utf-8").splitlines()
+
+    _file_content = remove_all_memory_flags(_file_content)
+    set_memory_flags(_file_content, _memory_min, _memory_max)
+    print_vm_memory_usage_during_start(_file_content)
```

Comment:
Instead of a dedicated one-liner function I'd prefer to do the 
```suggestion
    file_content.append("-XshowSettings:vm"). # print the JVM memory options during start
```

Reply:
Same comments as above, for me that makes the main function more verbose and less readable.